src/token/fa2.mligo

```diff
@@ -128,22 +128,22 @@ let balance_of (params, store : balance_request_params * storage): return =
 // Update operators entrypoint
 // -----------------------------------------------------------------
 [@inline]
-let validate_operator_token (token_id, frozen_token_id, is_operator_admin : token_id * token_id * bool): token_id =
-  if (token_id = frozen_token_id) then
+let validate_operator_token (token_id, store, is_operator_admin : token_id * storage * bool): token_id =
```

Comment:
Ok so the:
```
it seems that Michelson won't allow always failing code, so have included this branch
```
in `validate_operator_token` was due to the fact that there was no other valid `token_id` aside from `frozen_token_id` *and*  that was an `OPERATION_PROHIBITED` unless the operator is `admin` (which is always authorized to operate them, so doesn't actually need to be an operator explicitly).

In other terms, these two things make the check absurd, at least until this PR comes along.

Now that there are (potentially) more valid `token_id` we can revert this code and make it check that:
- for `frozen_token_id` no operator is valid (`admin` technically is, but doesn't need the `update_operators` call)
- for every other id, it must be one of the allowed ones

So I think this is the correct implementation now is:
```ocaml
let validate_operator_token (token_id, store : token_id * storage): token_id =
  if (token_id = frozen_token_id)
  then (failwith("OPERATION_PROHIBITED") : token_id)
  else validate_token_type (token_id, store)
```

Reply:
> So I think this is the correct implementation now is:

Agree.